Sources/ParseLiveQuery/Client.swift

```diff
@@ -149,13 +150,13 @@ extension Client {
             )
             subscriptions.append(subscriptionRecord)
 
-            if socket == nil {
-                if !disconnected {
-                    reconnect()
-                }
-            } else {
-                sendOperationAsync(.Subscribe(requestId: subscriptionRecord.requestId, query: query))
-            }
+        if socket == nil || socket?.readyState == .CLOSED{
+            userDisconnected = false
```

Comment:
Thats a good idea. Although I think to do this right, at some point we should either create a socket class(that perhaps sub-classes SRWebSocket) or a struct just for the socket itself. Then we would have a lot more options about what is private or public, and allow the socket to manage itself. This would also put us in the position to let the the primary delegate private(make ClientPrivate actually private) and let us create our own public delegate that removes all the SRWebSocketDelegate functions from public view.

For now lets simply to add a check for it. Speaking of which. I have a quick question involving explicit disconnect and then adding a resubscribe. I was thinking of printing 

> Warning: The client was explicitly disconnected! You must explicitly call .reconnect() in order to process your subscriptions.

Also, it probably wouldn't hurt to add to the documentation that a explicit reconnect requires a explicit reconnect.